app/src/main/java/org/oppia/android/app/mydownloads/downloads/DownloadsAccessDialogFragmentPresenter.kt

```diff
@@ -0,0 +1,106 @@
+package org.oppia.android.app.mydownloads.downloads
+
+import android.app.Dialog
+import android.view.KeyEvent
+import android.view.inputmethod.EditorInfo
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.fragment.app.Fragment
+import org.oppia.android.R
+import org.oppia.android.app.fragment.FragmentScope
+import org.oppia.android.app.utility.TextInputEditTextHelper.Companion.onTextChanged
+import org.oppia.android.app.viewmodel.ViewModelProvider
+import org.oppia.android.databinding.DownloadsAccessDialogBinding
+import javax.inject.Inject
+
+/** The presenter for [DownloadsAccessDialogFragment]. */
+@FragmentScope
+class DownloadsAccessDialogFragmentPresenter @Inject constructor(
+  private val activity: AppCompatActivity,
+  private val fragment: Fragment,
+  private val viewModelProvider: ViewModelProvider<DownloadsAccessViewModel>
+) {
+
+  private val downloadsTopicDeleteInterface =
+    fragment.parentFragment as DownloadsTopicDeleteInterface
+  private val downloadsAccessAllowedViewModel by lazy {
+    getDownloadsAccessViewModel()
+  }
+
+  fun handleOnCreateDialog(
+    adminPin: String?,
+    allowDownloadAccess: Boolean
+  ): Dialog {
+    val binding: DownloadsAccessDialogBinding =
+      DataBindingUtil.inflate(
+        activity.layoutInflater,
+        R.layout.downloads_access_dialog,
+        /* parent= */ null,
+        /* attachToParent= */ false
+      )
+
+    binding.apply {
+      lifecycleOwner = fragment
+      viewModel = downloadsAccessAllowedViewModel
+    }
+
+    // [onTextChanged] is a extension function defined at [TextInputEditTextHelper]
+    binding.downloadsAccessInputPinEditText.onTextChanged { confirmPin ->
+      confirmPin?.let {
+        if (
+          downloadsAccessAllowedViewModel.errorMessage.get()
+            ?.isNotEmpty()!! && downloadsAccessAllowedViewModel.inputPin.get() == it
+        ) {
+          downloadsAccessAllowedViewModel.inputPin.set(it)
+        } else {
+          downloadsAccessAllowedViewModel.inputPin.set(it)
+          downloadsAccessAllowedViewModel.errorMessage.set("")
+        }
+      }
+    }
+
+    val dialog = AlertDialog.Builder(fragment.requireContext(), R.style.AlertDialogTheme)
+      .setTitle(R.string.downloads_access_dialog_heading)
+      .setView(binding.root)
+      .setMessage(R.string.downloads_access_dialog_message)
+      .setPositiveButton(R.string.admin_settings_submit, null)
```

Comment:
@rt4914 should we create a single submit string to use everywhere, or should I re-create a string with the same value but id specific to here?

Reply:
Also, we can brief all such similar information and keep in our wiki of coding style.

Reply:
We should keep all these separate. 
Reason: Though its written as `Submit` in english at all places but it might happen that in other languages we might use multiple words as replacement of `Submit` button depending upon context.

Reply:
done